### PR TITLE
UOE-9681,UOE-9670: Add OWS hosted PBS vanilla endpoint, for prebid.js DIY client

### PR DESCRIPTION
**Pull request by AvinashKapre**
# Description
**Endpoint**: http://localhost:8000/openrtb2/auction?source=pbjs

JIRA: 

1. https://inside.pubmatic.com:9443/jira/browse/UOE-9681
2. https://inside.pubmatic.com:9443/jira/browse/UOE-9670

### CODE REVIEW

File: modules/pubmatic/openwrap/beforevalidationhook.go [hunk 1]
```diff
@@ -25,14 +26,13 @@ func (m OpenWrap) handleBeforeValidationHook(
 	moduleCtx hookstage.ModuleInvocationContext,
 	payload hookstage.BeforeValidationRequestPayload,
 ) (hookstage.HookResult[hookstage.BeforeValidationRequestPayload], error) {
-	result := hookstage.HookResult[hookstage.BeforeValidationRequestPayload]{
-		Reject: true,
-	}
+	result := hookstage.HookResult[hookstage.BeforeValidationRequestPayload]{}
```
Comment (pm-nilesh-chate) [inline modules/pubmatic/openwrap/beforevalidationhook.go:29]: add 
```
if hybrid-endpoint {
reject = false
return
}
```

File: modules/pubmatic/openwrap/beforevalidationhook.go [hunk 2]
```diff
@@ -844,3 +849,18 @@ func getPubID(bidRequest openrtb2.BidRequest) (pubID int, err error) {
...
+	if imp.TagID == "" {
+		tagId = impExt.Data.PbAdslot
+	}
+	if impExt.Gpid != "" {
+		tagId = impExt.Gpid
+		if idx := strings.Index(impExt.Gpid, "#"); idx != -1 {
+			tagId = impExt.Gpid[:idx]
+		}
+	}
+	return tagId
```
Comment (pm-nilesh-chate) [inline modules/pubmatic/openwrap/beforevalidationhook.go:865]:
```
if impExt.Gpid != "" {
         if idx := strings.Index(impExt.Gpid, "#"); idx != -1 {
                  return impExt.Gpid[:idx]
         }
} else if imp.TagID != "" {
      return imp.TagID
}

return impExt.Data.PbAdslot
```

File: modules/pubmatic/openwrap/auctionresponsehook.go [hunk 3]
```diff
@@ -39,6 +39,10 @@ func (m OpenWrap) handleAuctionResponseHook(
 		m.metricEngine.RecordPublisherResponseTimeStats(rctx.PubIDStr, int(time.Since(time.Unix(rctx.StartTime, 0)).Milliseconds()))
 	}()
 
+	if rctx.Endpoint == models.EndpointOWS2S {
+		return result, nil
```
Comment (pm-nilesh-chate) [inline modules/pubmatic/openwrap/auctionresponsehook.go:43]: you need to run tracker Mutation call (refractor accordingly)

Comment (AvinashKapre) [reply modules/pubmatic/openwrap/auctionresponsehook.go:43]: Will adress in logger,tracker PR.

**Review by pm-nilesh-chate — APPROVED** [review]

**Review by pm-nilesh-chate — APPROVED** [review]

**Review by pm-saurabh-narkhede — APPROVED** [review]
LGTM

**Pull request merged**